contracts/oracles/WstEthOracle.sol

```diff
@@ -0,0 +1,65 @@
+// SPDX-License-Identifier: bsl-1.1
+
+pragma solidity 0.7.6;
+
+import "../helpers/SafeMath.sol";
+import "../helpers/ERC20Like.sol";
+import "../interfaces/IWstEthToken.sol";
+import "../interfaces/IStEthPriceFeed.sol";
+import "../interfaces/IOracleUsd.sol";
+import "../interfaces/IOracleRegistry.sol";
+import "../interfaces/IOracleEth.sol";
+import "../VaultParameters.sol";
+
+/**
+ * @title WstEthOracle
+ * @dev Wrapper to quote wstETH ERC20 token that represents the account's share of the total supply of stETH tokens. https://docs.lido.fi/contracts/wsteth/
+ **/
+
+contract WstEthOracle is IOracleUsd, Auth  {
+    using SafeMath for uint;
+
+    IOracleRegistry public immutable oracleRegistry;
+
+    uint immutable stEthDecimals;
+
+    address public stEthPriceFeed;
+
+    address public WETH;
```

Comment:
Since  constructor get address `WETH = IOracleRegistry(_oracleRegistry).WETH();` compiler does not give set WETH immutable, but possible set WETH as immutable if  add argument to constructor parametrs